gnocchi/rest/aggregates/processor.py

```diff
@@ -167,16 +169,34 @@ def aggregated(timeseries, aggregation, from_timestamp=None,
                     'timespan (%.2f%%)' % (needed_percent_of_overlap,
                                            percent_of_overlap))
 
-        if aggregation in AGG_MAP:
-            values = AGG_MAP[aggregation](values, axis=1)
-        elif aggregation == 'count':
-            values = numpy.count_nonzero(~numpy.isnan(values), axis=1)
+        try:
+            granularity, times, values, is_aggregated = (
+                agg_operations.evaluate(operations, key, times, values,
+                                        False, references[key]))
+        except agg_operations.InvalidOperations as e:
+            raise UnAggregableTimeseries("Invalid operations: %s" % e.reason)
+
+        # NOTE(sileht): Check if an operations have aggregated the result.
+        # (aggregate, or binary operation between metrics, ...)
+        # Otherwise we tag each resulting serie with its original reference.
+        if len(values.shape) != 2:
+            raise UnAggregableTimeseries("Unexpected resulting array shape")
+        elif is_aggregated:
+            if values.shape[1] != 1:
+                raise UnAggregableTimeseries("Unexpected resulting "
```

Comment:
IIUC this kind of exception cannot be raised by a user providing wrong data. So if it happens, it's a bug. I don't think we should use `UnAggregableTImeseries` because that would be caught by the `api.py` code and returned as a 400. We probably want a 500 here and a traceback. So we need a different exception type, something like a `RuntimeError`.

Reply:
Right, fixed.